Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/ui/create/ProfileCreateFragment.kt

```diff
@@ -82,39 +81,13 @@ class ProfileCreateFragment : Fragment(R.layout.profile_create_fragment), AutoIn
 
             // Phone
             phoneInputEdit.doAfterTextChanged {
-                // Propagate phone number to view model if it matches the pattern
-                if (Patterns.PHONE.matcher(it.toString()).matches()) {
-                    viewModel.phoneChanged(it.toString())
-                } else {
-                    viewModel.phoneChanged("")
-                }
-            }
-            phoneInputEdit.setOnFocusChangeListener { _, hasFocus ->
-                // Validate phone number
-                if (!hasFocus && !Patterns.PHONE.matcher(phoneInputEdit.text.toString()).matches()) {
-                    phoneInputLayout.error = root.resources.getString(R.string.rat_profile_create_phone_error)
-                } else {
-                    phoneInputLayout.error = null
-                }
+                viewModel.phoneChanged(it.toString())
             }
             phoneInputEdit.addTextChangedListener(PhoneNumberFormattingTextWatcher())
 
             // E-mail
             emailInputEdit.addEmojiFilter().doAfterTextChanged {
-                // Propagate email to view model if it matches the pattern
-                if (Patterns.EMAIL_ADDRESS.matcher(it.toString()).matches()) {
```

Comment:
Shouldn't we keep the e-mail validation at least, to have a valid e-mail? When I scan the QR from the profile it also displays the E-Mail field. Maybe hospitals/officials would use this

Reply:
Alright, I will keep the validation for the email.

Reply:
Updated the validation. In the end I also added it for the phone number as it wasn't much trouble.